File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        std::vector<int> result;
        std::queue<TreeNode*> q;
        q.push(root);

        while (!q.empty()) {
            TreeNode* right_side = nullptr;
            int size = q.size();

            for (int i = 0; i < size; i++) {
                TreeNode* node = q.front();
                q.pop();
                if (node) {
                    right_side = node;
                    q.push(node->left);
                    q.push(node->right);
                }
            }
            if (right_side) { result.push_back(right_side->val); }
        }

        return result;
    }
};

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []

        def _traverse(node):
            if not node:
                return 
                
            nonlocal res

            _traverse(node.left)
            res.append(node.val)
            _traverse(node.right)

        _traverse(root)

        return res